pkg/test/step.go

```diff
@@ -414,7 +414,12 @@ func (s *Step) Run(namespace string) []error {
 		return testErrors
 	}
 	for _, collector := range s.Assert.Collectors {
+		collector.Validate()
 		s.Logger.Logf("collecting log output for %s", collector.String())
+		if !collector.Valid || collector.Command() == nil {
+			s.Logger.Log("skipping invalid assertion collector")
```

Comment:
Please make this a fatal error. I find it extremely frustrating to find that the error handling code does not work exactly when I need it most to debug an elusive flake that just occured.

Reply:
I can't see that this would make sense...   This is after all the tests have run.. and if we are here at all... the test is already failed... collectors are only run if there is a failure.

Reply:
@porridge the frustration is understandable...   as I think more about it... there is nothing more we can do here.
1. the testsuite is already marked as failed
2. there will be an entry in the log above for the configuration.
3. that log will have a good lead on what the issue is... i.e., `command collector requires a command`

Reply:
Oh, I didn't realize we validate the config *after* we act on it!
How about validating at the beginning, before running anything?